wgpu/backends/wgpu_native/_api.py

```diff
@@ -2735,6 +2724,66 @@ def begin_render_pass(
         encoder._objects_to_keep_alive = objects_to_keep_alive
         return encoder
 
+    # Pulled out from create_render_pass because it was too large.
+    @staticmethod
+    def _create_c_render_pass_depth_stencil_attachment(ds_attachment):
+        view = ds_attachment["view"]
+        depth_read_only = stencil_read_only = False
+        depth_load_op = depth_store_op = stencil_load_op = stencil_store_op = 0
+        depth_clear_value = stencil_clear_value = 0
+        depth_keys_okay = stencil_keys_okay = False
+        # All depth texture formats have "depth" in their name.
+        if "depth" in view.texture.format:
+            depth_read_only = ds_attachment.get("depth_read_only", False)
+            if not depth_read_only:
+                depth_keys_okay = True
+                depth_load_op = ds_attachment["depth_load_op"]
+                depth_store_op = ds_attachment["depth_store_op"]
+                if depth_load_op == "clear":
+                    depth_clear_value = ds_attachment["depth_clear_value"]
+        # All stencil texture formats all have "stencil" in their name
+        if "stencil" in view.texture.format:
+            stencil_read_only = ds_attachment.get("stencil_read_only", False)
+            if not stencil_read_only:
+                stencil_keys_okay = True
+                stencil_load_op = ds_attachment["stencil_load_op"]
+                stencil_store_op = ds_attachment["stencil_store_op"]
+                # We only need this if load_op == "clear". It has a default value.
+                stencil_clear_value = ds_attachment.get("stencil_clear_value", 0)
+
+        # By the spec, we shouldn't allow "depth_load_op" or "depth_store_op"
+        # unless we have a non-read-only depth format. Likewise, "stencil_load_op"
+        # and "stencil_store_op" aren't allowed unless we have a non-read-only stencil
+        # format.
+        # But until now, they were required, even if not needed. So let's make it a
+        # warning for now, with the possibility of making it an error in the future.
+        unexpected_keys = [
+            *(("depth_load_op", "depth_store_op") if not depth_keys_okay else ()),
+            *(("stencil_load_op", "stencil_store_op") if not stencil_keys_okay else ()),
+        ]
+        for key in unexpected_keys:
+            if key in ds_attachment:
+                warnings.warn(
+                    f"Unexpected key {key} in depth_stencil_attachment",
+                    DeprecationWarning,
+                    stacklevel=2,
+                )
```

Comment:
We use `logger.warning()` instead of the warnings module.

Another thing is that (I think) this warning would be shown on each draw, which may be a bit much. Perhaps we can do something like:
```py
if not getattr(self._device, "_showed_depth_stencilwarning1", False):
    self._device._showed_depth_stencilwarning1 = True
    logger.warning(....)
```